src/Network/Kontiki/Raft/Candidate.hs

```diff
@@ -123,14 +123,15 @@ handleElectionTimeout = do
                                  }
 
 -- | Handles `HeartbeatTimeout'.
-handleHeartbeatTimeout :: (Functor m, Monad m)
+handleHeartbeatTimeout :: (Monad m)
                        => TimeoutHandler HeartbeatTimeout a Candidate m
 handleHeartbeatTimeout = do
+    resetHeartbeatTimeout
```

Comment:
Are you sure this is the correct thing to do? We don't have to send a heartbeat as long as we're Candidate, so why resetting its timeout?